### DISCUSSION TITLE
Decompiling of inner classes fails. Non existant.

### DISCUSSION THREAD

**Discussion by tipa40** [mfussenegger/nvim-jdtls#545] [Q&A]
Hi,
When I decompile an library class via vim.lsp.buf.definition(), if that class is containing an inner class it will now show up in the decompiled class, just as if it does not exist. Are there any suggestion on what I can do about this? It seems I am using FernFlower decompiler which is the default in my build. I am using the current version of nvim-jdtls at this point in time. Is there any better alternative to FernFlower available for me to configure or is this an issue with nvim-jdtls so that I should add an Issue?

**Comment by mfussenegger** [comment]
eclipse.jdt.ls switched to use the fernflower decompiler by default in https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2704

So this is either a eclipse.jdt.ls issue or a fernflower issue. Nothing to do with nvim-jdtls. It's only getting the content from the server. _What_ content is returned, is up to the server.

**Reply by tipa40** [reply]
Ok thanks 👍🏻😊
